molgenis-js/src/main/java/org/molgenis/js/magma/JsMagmaScriptEvaluator.java

```diff
@@ -73,16 +73,39 @@ public Object eval(String expression, Entity entity)
 		return value;
 	}
 
-	private static Map<String, Object> toScriptEngineValueMap(Entity entity)
+	/**
+	 * Convert entity to a JavaScript object.
+	 * Adds "_idValue" as a special key to every level for quick access to the id value of an entity.
+	 *
+	 * @param entity The entity to be flattened, should start with non null entity
+	 * @return flattened entity as key value map
+	 */
+	private Object toScriptEngineValueMap(Entity entity, int depth)
 	{
-		Map<String, Object> map = Maps.newHashMap();
-		entity.getEntityType()
-			  .getAtomicAttributes()
-			  .forEach(attr -> map.put(attr.getName(), toScriptEngineValue(entity, attr)));
-		return map;
+		if (entity != null)
+		{
+			Object idValue = toScriptEngineValue(entity, entity.getEntityType().getIdAttribute(), 0);
+			if (depth == 0)
+			{
+				return idValue;
+			}
+			else
+			{
+				Map<String, Object> map = Maps.newHashMap();
+				entity.getEntityType()
+					  .getAtomicAttributes()
+					  .forEach(attr -> map.put(attr.getName(), toScriptEngineValue(entity, attr, depth)));
+				map.put("_idValue", idValue);
```

Comment:
should '_idValue' be added to the help pages as reserved keyword and or blacklisted as prop name ?

Reply:
Yes